### DISCUSSION TITLE
GAP usage/function FAQ

### DISCUSSION THREAD

**Discussion by Jeffreyke** [gap-system/gap#4592] [Q&A] [answered]
I wish someone can give me a hand on following application problems:
(1) How to get quotient Ring:
E.g. 
```
R :=Integers;
Integers
gap> S :=Ideal(R, [17]);
<two-sided ideal in Integers, (1 generators)>
gap> R_S := R/S;
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method .......
```
I couldn't build Quotient Ring like Quotient Group

(2) Why Factors() doesn't work as expected  on GaussianIntegers Ring ?
```
x :=X(GaussianIntegers, "x");
gap> Factors(x^2+1);
[ x^2+1 ]
```
Based on Ring coefficients definition, Factors(x^2+1) should get:
[x - i, x+i]
Right?

(3) Can I extract a Group structure from some existing Ring for its multiplication group? 
```
 gap> z79 := Integers mod 79;
GF(79)
gap> IsField(z79);
true
gap> StructureDescription(z79);
Error, no method found! For debugging hints type ? ....
```
How can I analyze some Ring's multiplication group independently?

**Comment by fingolfin** [comment, marked as answer]
Hi there!

First, a meta remark: I took the liberty of editing your post a bit, it was very difficult to read due to your user of `=====` as separators, which causes text to render in BIG BOLD LETTERS -- I've replaced those by triple backticks, which exist to render code, and the result looks better.

Regarding your first question: To get `Z/nZ`, just write `Integers mod n`. For primes, though, GAP will by default return the finite field `GF(p)`:
```
gap> Integers mod 17;
GF(17)
gap> Integers mod 17^2;
(Integers mod 289)
```

Regarding your second question: the variable `x` does not really store the coefficient domain. You need to pass the polynomial ring in which you want to factor to the `Factors` function (its manual entry explains that and gives an example). Finally, GAP really only knows how to factor over coefficient domains which are *fields*. So try this:
```
gap> R:=PolynomialRing(GaussianRationals,["x"]);
GaussianRationals[x]
gap> x:=R.1;
x
gap> Factors(R, x^2+1);
[ x+(-E(4)), x+E(4) ]
```

Regarding the third point:
```
gap> z79 := Integers mod 79;
GF(79)
gap> StructureDescription(Units(z79));
"C78"
```

But of course we knew that: the unit group of a finite field is always a cyclic group.
